extensions/tasks/NLUTestV0/publishTestResults.ts

```diff
@@ -0,0 +1,50 @@
+import * as tl from 'azure-pipelines-task-lib/task';
+
+type TestRunner = "JUnit" | "NUnit" | "VSTest" | "XUnit" | "CTest";
+
+const TESTRUN_SYSTEM = 'VSTS - PTR';
+
+function isNullOrWhitespace(input: any) {
+    if (typeof input === 'undefined' || input === null) {
+        return true;
+    }
+    return input.replace(/\s/g, '').length < 1;
+}
+
+function publish(testRunner: TestRunner, resultFiles: string[], testRunSystem: string) {
+    var properties: { [key: string]: string | string[] } = {};
+    properties['type'] = testRunner;
+
+    if (resultFiles) {
+        properties['resultFiles'] = resultFiles;
+    }
+    properties['testRunSystem'] = testRunSystem;
+
+    tl.command('results.publish', properties, '');
+}
+
+export async function publishTestResults(testResultsFiles: string[], searchFolder: string, testRunner: TestRunner = 'NUnit') {
+
+    if (isNullOrWhitespace(searchFolder)) {
```

Comment:
I have a function for the search folder in `index.ts`: `getCompareOutputPath()`.

It returns the `compareOutput` input if set, otherwise it will be `$(Agent.TempDirectory)/.nlu` if `publishTestResults` or `publishNLUResults` is set.